ros2/src/airsim_ros_pkgs/CMakeLists.txt

```
@@ -0,0 +1,163 @@
+#   DEPENDENCIES builtin_interfaces nav_msgs geographic_msgs std_srvs
+#   tf2_sensor_msgs geometry_msgs tf2_geometry_msgs sensor_msgs mavros_msgs std_msgs)
+
+
+
+set(AIRSIM_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../../../)
+
+add_subdirectory("${AIRSIM_ROOT}/cmake/rpclib_wrapper" rpclib_wrapper)
+add_subdirectory("${AIRSIM_ROOT}/cmake/AirLib" AirLib)
+add_subdirectory("${AIRSIM_ROOT}/cmake/MavLinkCom" MavLinkCom)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CXX_EXP_LIB
+  "-nostdinc++ -I/usr/include/c++/8 -I/usr/include/x86_64-linux-gnu/c++/8 -nodefaultlibs 
+-l/usr/lib/x86_64-linux-gnu/libc++.so -l/usr/lib/x86_64-linux-gnu/libc++abi.so
+-lm -lc -lgcc_s -lgcc -g    
+-lstdc++fs -fmax-errors=10 -Wnoexcept -Wstrict-null-sentinel")
+set(RPC_LIB_INCLUDES " ${AIRSIM_ROOT}/external/rpclib/rpclib-2.3.0/include")
+set(RPC_LIB rpc)
+message(STATUS "found RPC_LIB_INCLUDES=${RPC_LIB_INCLUDES}")
+
+set(INCLUDE_DIRS include ${rclcpp_INCLUDE_DIRS} ${nav_msgs_INCLUDE_DIRS}
+  ${geographic_msgs_INCLUDE_DIRS} ${std_srvs_INCLUDE_DIRS} ${tf2_ros_INCLUDE_DIRS}
+  ${tf2_sensor_msgs_INCLUDE_DIRS} ${rclpy_INCLUDE_DIRS} ${tf2_INCLUDE_DIRS}
+  ${image_transport_INCLUDE_DIRS} ${geometry_msgs_INCLUDE_DIRS}
+  ${cv_bridge_INCLUDE_DIRS} ${tf2_geometry_msgs_INCLUDE_DIRS}
+  ${sensor_msgs_INCLUDE_DIRS} ${mavros_msgs_INCLUDE_DIRS}
+  ${rosidl_default_generators_INCLUDE_DIRS} ${ament_cmake_INCLUDE_DIRS}
+  ${std_msgs_INCLUDE_DIRS}
+  ${AIRSIM_ROOT}/AirLib/deps/eigen3
+  ${AIRSIM_ROOT}/AirLib/include
+  ${RPC_LIB_INCLUDES}
+  ${AIRSIM_ROOT}/MavLinkCom/include
+  ${AIRSIM_ROOT}/MavLinkCom/common_utils
+  ${OpenCV_INCLUDE_DIRS}
+  ${Boost_INCLUDE_DIRS}
+  ${airsim_interfaces_INCLUDE_DIRS})
+include_directories(${INCLUDE_DIRS})
+
+set(LIBRARY_DIRS ${rclcpp_LIBRARY_DIRS} ${nav_msgs_LIBRARY_DIRS}
+  ${geographic_msgs_LIBRARY_DIRS} ${std_srvs_LIBRARY_DIRS} ${tf2_ros_LIBRARY_DIRS}
+  ${tf2_sensor_msgs_LIBRARY_DIRS} ${rclpy_LIBRARY_DIRS} ${tf2_LIBRARY_DIRS}
+  ${image_transport_LIBRARY_DIRS} ${geometry_msgs_LIBRARY_DIRS}
+  ${cv_bridge_LIBRARY_DIRS} ${tf2_geometry_msgs_LIBRARY_DIRS}
+  ${sensor_msgs_LIBRARY_DIRS} ${mavros_msgs_LIBRARY_DIRS}
+  ${rosidl_default_generators_LIBRARY_DIRS} ${ament_cmake_LIBRARY_DIRS}
+  ${std_msgs_LIBRARY_DIRS}
+  ${airsim_interfaces_LIBRARY_DIRS})
+
+link_directories(${LIBRARY_DIRS})
+
+set(LIBS ${rclcpp_LIBRARIES} ${nav_msgs_LIBRARIES} ${geographic_msgs_LIBRARIES}
+  ${std_srvs_LIBRARIES} ${tf2_ros_LIBRARIES} ${tf2_sensor_msgs_LIBRARIES}
+  ${rclpy_LIBRARIES} ${tf2_LIBRARIES} ${image_transport_LIBRARIES}
+  ${geometry_msgs_LIBRARIES} ${cv_bridge_LIBRARIES} ${tf2_geometry_msgs_LIBRARIES}
+  ${sensor_msgs_LIBRARIES} ${mavros_msgs_LIBRARIES}
+  ${rosidl_default_generators_LIBRARIES} ${ament_cmake_LIBRARIES}
+  ${std_msgs_LIBRARIES}
+  ${airsim_interfaces_LIBRARIES})
+
+find_package(Boost REQUIRED)
+find_package(OpenCV REQUIRED)
+
+add_library(airsim_settings_parser src/airsim_settings_parser.cpp)
+target_link_libraries(airsim_settings_parser ${LIBS} AirLib)
```

I don't think anything in `LIBS` is required for this library.
```suggestion
target_link_libraries(airsim_settings_parser AirLib)
```

---

fixed